src/jit/lclvars.cpp

```diff
@@ -3809,6 +3809,11 @@ void Compiler::lvaMarkLclRefs(GenTreePtr tree)
     allowStructs = varTypeIsStruct(varDsc);
```

Comment:
@CarolEidt To be consistent with the comment and not allow mismatched TYP_SIMD*/TYP_STRUCT to  pass.  XARCH should make this line `allowStructs = varTypeIsStruct(varDsc) && (tree->gtType == TYP_BYREF);